feat: request verification email

dcustodio opened the pull request:
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

_What this PR does_

Adds a new banner to allow users to (re)send the verification email. 
includes the work done on #3229

## Git Issues

Closes #3249

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

![](https://private-user-images.githubusercontent.com/2907004/301838196-16fa7003-d18a-4242-95b1-37a63d9d801a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyMzE2MDYsIm5iZiI6MTcwNzIzMTMwNiwicGF0aCI6Ii8yOTA3MDA0LzMwMTgzODE5Ni0xNmZhNzAwMy1kMThhLTQyNDItOTViMS0zN2E2M2Q5ZDgwMWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDZUMTQ1NTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmZkZDA5ZmVjOGUyYjQ2MTBhZjQxZTE2MDdiMjg1ZjVhYTAzYzM5N2QxMzA3ZWUwMTkyODM3NjRjMmJjNjk5MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.iwfxVCAfexNKCMttZ9fbFT0fst-l9D3CyD0pidUnN7E)
---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

davehakkens commented:
Looks/works nice @dcustodio!
What about we add a backup in the banner in case the email sender didnt work, like the current bug.


Perhaps extend the message like this:
"Verification email sent. Please check your inbox and spam folder. (Let us [know](mailto:platform@onearmy.earth) if you didnt get it)"

<img width="414" alt="afbeelding" src="https://github.com/ONEARMY/community-platform/assets/13672737/0cc6f15c-8f09-450f-8f05-5547c7277e64">

dcustodio commented:
> Looks/works nice @dcustodio! What about we add a backup in the banner in case the email sender didnt work, like the current bug.
> 
> Perhaps extend the message like this: "Verification email sent. Please check your inbox and spam folder. (Let us [know](mailto:platform@onearmy.earth) if you didnt get it)"
> 
> <img alt="afbeelding" width="414" src="https://private-user-images.githubusercontent.com/13672737/302762550-0cc6f15c-8f09-450f-8f05-5547c7277e64.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyNTExMzMsIm5iZiI6MTcwNzI1MDgzMywicGF0aCI6Ii8xMzY3MjczNy8zMDI3NjI1NTAtMGNjNmYxNWMtOGYwOS00NTBmLThmMDUtNTU0N2M3Mjc3ZTY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA2VDIwMjAzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3MmNkYzA4YmMyM2M5Y2QwZjI4ZTQ1Y2FlOTZhZWNlNzZmMDNkMTk5ZjM0Mzg3ZTYyYWIxNzUwZTMwYTI4ODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.c7IGXTX0Kr5ged7_UbfyhAzFOfOfnnR73jKqjcW2yss">

Good idea, I updated the PR with those suggestions, I added some generic error message but it's up for debate.

https://github.com/ONEARMY/community-platform/assets/2907004/f1e0711f-0114-4bc0-a041-e80e3c7ac065

![Screenshot 2024-02-08 124826](https://github.com/ONEARMY/community-platform/assets/2907004/e1f5d401-6ef3-4f38-8944-e14043cab050)

benfurber reviewed (changes requested):
This is a really nice bit of work, thank you.

A few minor comments left, plus the below please.

Rather than all of #3229, can you please just duplicate the store relayed changes - we don't want the functions work in master yet.

davehakkens commented on src/common/Alerts/AlertProfileVerification.tsx:
```diff
@@ -0,0 +1,84 @@
+            (Let us{' '}
+            <ExternalLink
+              sx={{ textDecoration: 'underline', color: 'grey' }}
+              href="mailto:platform@onearmy.earth"
```
Good catch. But would actually be useful to go to platform@onearmy.earth so it has its own feedback stream that can go quicker to dev. But also dont mind to much to use hello@onearmy.earth if it makes the code cleaner.

dcustodio replied:
I've kept the `platform@onearmy` email but added the subject.  Happy to change to the `hello@` email if I misunderstood the consensus. 😅

dcustodio commented on src/common/Alerts/AlertProfileVerification.tsx:
```diff
@@ -0,0 +1,84 @@
+  const isVerificationSuccessful = verificationState === 'sent'
+  const isVerificationPending = verificationState === 'pending'
+  const alertLabel = isVerificationPending
+    ? '✉️ Click here to receive an email to confirm your account.'
```
Sure, better that way! I've added the Icon

dcustodio commented on src/common/Alerts/AlertProfileVerification.tsx:
```diff
@@ -0,0 +1,84 @@
+  const isVerificationPending = verificationState === 'pending'
+  const alertLabel = isVerificationPending
+    ? '✉️ Click here to receive an email to confirm your account.'
+    : "Sorry, we couldn't send an email. Please try again later."
```
Sounds like a plan! Let's add it afterwards 👍

dcustodio commented on src/common/Alerts/AlertProfileVerification.tsx:
```diff
@@ -0,0 +1,84 @@
+              sx={{ textDecoration: 'underline', color: 'grey' }}
+              href="mailto:platform@onearmy.earth"
+            >
+              know
```
I've moved these sentences into two `const` like suggested 👍

dcustodio commented on src/common/Alerts/AlertProfileVerification.tsx:
```diff
@@ -0,0 +1,84 @@
+            Verification email sent. Please check your inbox and spam folder.
+            (Let us{' '}
+            <ExternalLink
+              sx={{ textDecoration: 'underline', color: 'grey' }}
```
It seems so. On that note, I've added grey color because I saw it elsewhere but it's up for debate.

benfurber reviewed (approved):
Awesome, thanks @dcustodio! Now please just rebase to master and I can get this merged in.

benfurber commented:
@all-contributors please add @dcustodio for contributing

benfurber commented:
https://github.com/all-contributors please add @dcustodio for code

benfurber commented:
@all-contributors please add @dcustodio for contributing

benfurber commented:
@all-contributors please add @dcustodio for code

benfurber commented:
@dcustodio The feature tests are failing because the changes to `src/stores/Message/message.store.tsx` you took from a branch of mine are still present in yours. Remove them and then this branch should be able to merge itself.
